src/routes/safe/components/Apps/utils.ts

```diff
@@ -200,3 +200,48 @@ export const getLegacyChainName = (chainName: string, chainId: string): string =
 
   return network
 }
+
+// Sanitize URL helper
+const invalidProtocolRegex = /^([^\w]*)(javascript|data|vbscript)/im
+const htmlEntitiesRegex = /&#(\w+)(^\w|;)?/g
+const ctrlCharactersRegex = /[\u0000-\u001F\u007F-\u009F\u2000-\u200D\uFEFF]/gim
+const urlSchemeRegex = /^([^:]+):/gm
+const relativeFirstCharacters = ['.', '/']
+
+function isRelativeUrlWithoutProtocol(url: string): boolean {
+  return relativeFirstCharacters.indexOf(url[0]) > -1
+}
+
+function decodeHtmlCharacters(str: string) {
+  return str.replace(htmlEntitiesRegex, (match, dec) => {
+    return String.fromCharCode(dec)
+  })
+}
+
+export function sanitizeUrl(url: string | null): string {
```

Comment:
I would suggest adding a few unit-tests for positive and negative cases.

Reply:
Added